### DISCUSSION TITLE
Suggestion: systemd service creator

### DISCUSSION THREAD

**Discussion by 102464** [cockpit-project/cockpit#16894] [Ideas]
As I know, Cockpit is a web-based graphical interface intended for everyone, including those who are new to linux. In my opinion, some improvements can be applied to make Cockpit better.
One improvement I want to suggest is a service creator in the "Services" tab. Usually when we want to create a new service, we need to write a service file manually and put it into /lib/systemd/system, which can only be done in terminal.
However, this is unconvenient and unfriendly for new linux users.
So if you can add a "create" button in Services tab, and make service creation process automated, that may be a big advancement, I think.

Like this:
![systemd1](https://user-images.githubusercontent.com/20787495/151709483-9c71ba47-62c4-4c32-b7b1-ad7937dc296b.PNG)
![systemd2](https://user-images.githubusercontent.com/20787495/151709499-c27219d8-1a88-43f2-b36d-dc2fa0ec1ba6.PNG)
(taken from https://mysystemd.talos.sh)

Would that be possible?

**Comment by martinpitt** [comment]
It's possible of course. systemd unit files support [literally hundreds of options](https://www.freedesktop.org/software/systemd/man/systemd.directives.html), so any UI can only support a small subset. Starting a `Type=simple` service with optional auto-restart on every boot would probably be a good start -- this is then roughly the complexity of the "create timer" dialog which we already have.

**Reply by garrett** [reply]
A simple one makes sense.

If we did try to tackle an advanced one, we'd probably have to do something like a key+value UI with the key being an autocomplete based off known directives. And possibly show the documentation section based on the selected key? 

But it probably wouldn't add too much value to people using Cockpit over editing a file with a `man` page open in another window.

I'd say we should try to go for a simple one that handles the most common tasks. 

Perhaps we could have an advanced expander for additional options, kind of like how we have the ability to add additional options to filesystem mounts. Except this would probably be in the form of a [PF Code Editor widget](https://www.patternfly.org/v4/components/code-editor) (basically a fancy `textarea`) instead of a text input or the key/value widgetry I suggested directly above.
